### DISCUSSION TITLE
Defense bot needs help (bugs and issues list)

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#14311] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

-only 2 can exist *total*, not per team, and it can't be exceeded via editor. they simply will stop spawning after the first 2, even if more are placed in editor. commands can spawn more ingame which functions normally.
-in pvp, they do not perceive the enemy team as enemies. they do not shoot, and can be [E] Play
-in pvp, only the coalition spawns with them if there is more than 1 spawned in a submarine. They should at least be split symmetrically between the teams.
-in the submarine editor, their inventory cannot be edited, so custom ammo is impossible without inserting it manually as a player. this adds tedium to using them in individual mission mode or (if fixed) pvp.
-in the submarine editor, they are hidden in menus and must be command spawned (unlike all other talent items, because it is technically a pet)
-they cannot fly up ladders, nor seem to stand on platforms, they will just fall with no way of returning, making them basically always sit in the ballast tank or bottom airlock or r29 cargo bay, whatever is the lowest point of the sub, they will eventually make their way to. having to constantly carry a *flying* robot is annoying. [this discussion mentions more specific things about it](https://github.com/FakeFishGames/Barotrauma/discussions/12207)
-the follow function cannot be toggled off. It will ALWAYS be in the way, and has to be locked behind a door or thrown in the ballast to not be in the way.
-the AI allows it to shoot the player on accident many times, it should not shoot if the player is going to be hit. It does pretty well, but there are times where it will move at the same pace in the same direction as the player, shooting the player in the face 20 times.  (e.g. running away from a monster, the bot strafes backwards at the same pace as you)
-it is possible to attempt to craft more than 2 defense bots, which will consume the materials, but not produce a bot. the recipe should be locked when the max is reached, then unlocked once one is destroyed.

### Reproduction steps

I attempted to be as descriptive as possible in the bugs, but in case it wasn't clear,

bug 1- spawn more than 2 defense bots in the sub editor, and see that only 2 spawn. 
bug 2- enter pvp, find the other team's defense bot, walk up to it, don't get shot
bug 3- enter pvp using a submarine with more than 2 defense bots, and see that only the coalition gets them, instead of each team getting 2.
bug 4- spawn a defense bot in submarine editor, attempt to remove the ammo from its inventory lol
bug 5- try to find it in the item menu of submarine editor
bug 6- spawn some defense bots on the r29, and play a few rounds. with all of the bots running around, doors opening, it is free to just fall into the cargo hold and never come out.
bug 7- spawn some defense bots, press e, and then try to get them to leave you alone. 
bug 8- do a [higher difficulty] abandoned outpost mudraptor nest mission, spawn some defense bots, flood the outpost so that the bots can actually navigate without being carried, and try to beat the mission without being shot at by your own bots.
bug 9- fabricate 2 defense bots. name them bob and joe, so you can tell which bot is which. attempt to craft one more, and witness your smg, 2 titanium-aluminum alloy, and FPGA circuit disappearing. also take time to notice that bob and joe are the only 2 bots in the room, despite the materials being missing. where did they go???

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Created a ticket about this in #14322.

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next patch. Closing discussion.
